src/Libraries/Inscriptions/Envelope.ts

```diff
@@ -178,16 +178,28 @@ function getEnvelopeContent(data: EnvelopeData[]) {
   };
 }
 
-function getParent(data: EnvelopeData[]) {
-  const startIndex = data.indexOf(PARENT_TAG);
-  if (startIndex === -1) {
-    return undefined;
+function getParents(data: EnvelopeData[]) {
+  const parents = [];
+  let scanIndex = 0;
+
+  while (data.indexOf(PARENT_TAG, scanIndex) !== -1) {
+    const parentIndex = data.indexOf(PARENT_TAG, scanIndex);
+    const parent = getParent(data[parentIndex + 1]);
+    if (parent) {
+      parents.push(parent);
+    }
+    scanIndex += 1;
   }
-  const parent = data[startIndex + 1];
-  if (!parent || !isBuffer(parent)) {
+  return parents;
+}
+
+function getParent(parentData: EnvelopeData) {
+  if (!isBuffer(parentData)) {
     return undefined;
   }
-  return `${parent.reverse().toString("hex")}i0`;
+  const txid = parentData.reverse().toString("hex").slice(0, 64);
+  const inscription_index = parseInt(parentData.reverse().toString("hex").slice(64), 16);
```

Comment:
is this tested to be correct? 🤔  from the [handbook example](https://docs.ordinals.com/inscriptions/provenance.html). 
lets say the buffer data from witness is `0x1f1e1d1c1b1a191817161514131211100f0e0d0c0b0a09080706050403020100ff` which is supposed to be `000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1fi255`

doing 
```js
const parentData = Buffer.from("1f1e1d1c1b1a191817161514131211100f0e0d0c0b0a09080706050403020100ff","hex")
const txid = parentData.reverse().toString("hex").slice(0, 64); 
// will result in "ff000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e"
```
see how the inscriptionindex now is in the first 2 characters (`ff`)

Reply:
Ah good catch. It's not always first 2 characters, in the case there are more than 256 then it's 100, so first 3 characters. 

Will fix.

Reply:
yeah that too but the issue i mentioned here is that the "reverse" is reversing wrongly. because from the witness it is in the format for `<inscription_tx><inscription_index>` when you reverse this whole string it becomes `<inscription_index><inscription_tx>`. Instead what you want is to separate the two first so slice the first 64 characters then reverse each component so you can get back `<inscription_tx><inscription_index>`

Reply:
Yup, fixed!